### PR TITLE
[New Connector] ServiceNow: Configuration & Ping feature

### PR DESCRIPTION
**Pull request by parth-elastic**
#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference

### CODE REVIEW

File: connectors/sources/servicenow.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for ServiceNow.
+
+        Returns:
+            dict: Default configuration.
+        """
+        return {
+            "url": {
```
Comment (vidok) [inline connectors/sources/servicenow.py:71]: referring to [this](https://github.com/elastic/connectors-python/pull/537#discussion_r1129641667) discussion. Could you please remove remove "non-default" fields?

File: connectors/sources/tests/test_servicenow.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+
+def test_get_configuration(patch_logger):
+    # Setup
+    klass = ServiceNowDataSource
+
+    # Execute
+    config = DataSourceConfiguration(klass.get_default_configuration())
+
+    # Assert
+    assert config["username"] == "admin"
```
Comment (timgrein) [inline connectors/sources/tests/test_servicenow.py:61]: We may want to check all fields or what's the purpose of only testing `username`?

**Comment by artem-shelkovnikov** [comment]
@parth-elastic do you have any plans on this PR?

**Comment by parth-elastic** [comment]
We have deprioritized this connector. We would be picking this in the next contract 

cc @nirav-elastic @danajuratoni

File: connectors/sources/servicenow.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+
+                    fetched_response = await response.read()
+                    if fetched_response == b"":
+                        raise InvalidResponse(
+                            "Request hasn't processed from ServiceNow server"
+                        )
+                    elif (
+                        not response.headers["Content-Type"]
+                        .strip()
+                        .startswith("application/json")
```
Comment (tarekziade) [inline connectors/sources/servicenow.py:206]: what types are you covering? (also `strip()` is not required)

**Pull request closed without merging**